src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectStrategyService.java

```diff
@@ -34,18 +41,31 @@
     private final JenkinsIntLogger logger;
     private final String remoteTempWorkspacePath;
     private final JenkinsProxyHelper jenkinsProxyHelper;
+    private final JenkinsConfigService jenkinsConfigService;
 
-    public DetectStrategyService(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsProxyHelper, String remoteTempWorkspacePath) {
+    public DetectStrategyService(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsProxyHelper, String remoteTempWorkspacePath, JenkinsConfigService jenkinsConfigService) {
         this.logger = logger;
         this.jenkinsProxyHelper = jenkinsProxyHelper;
         this.remoteTempWorkspacePath = remoteTempWorkspacePath;
+        this.jenkinsConfigService = jenkinsConfigService;
     }
 
-    public DetectExecutionStrategy getExecutionStrategy(IntEnvironmentVariables intEnvironmentVariables, OperatingSystemType operatingSystemType, String remoteJdkHome) {
-        DetectExecutionStrategy detectExecutionStrategy;
+    public DetectExecutionStrategy getExecutionStrategy(IntEnvironmentVariables intEnvironmentVariables, OperatingSystemType operatingSystemType, String remoteJdkHome, DetectDownloadStrategy detectDownloadStrategy)
+        throws IntegrationException {
+        if (detectDownloadStrategy == null || detectDownloadStrategy instanceof InheritFromGlobalDownloadStrategy) {
+            DetectGlobalConfig detectGlobalConfig = jenkinsConfigService.getGlobalConfiguration(DetectGlobalConfig.class)
+                                                        .orElseThrow(() -> new DetectJenkinsException("Unable to identify Detect Global Configuration."));
+            detectDownloadStrategy = detectGlobalConfig.getDownloadStrategy();
+        }
+
+        logger.info("Running Detect using strategy: " + (detectDownloadStrategy != null ? detectDownloadStrategy.getClass().getSimpleName() : null));
```

Comment:
If the global configuration DownloadStrategy is null, it appears we default to the Script or Jar download strategy (since the instanceof will evaluate to false on line 66). We should make that clear in this log message.

Also, the class name isn't likely to mean anything to our customers. We probably want to log the DISPLAY_NAME field from those classes since it's what shows up in the UI